ios/Classes/SwiftFlutterDownloaderPlugin.swift

```diff
@@ -0,0 +1,207 @@
+import Flutter
+import UIKit
+import BackgroundTasks
+
+private enum DownloadStatus {
+  case running
+  case completed
+  case failed
+  case canceled
+  case paused
+}
+
+private class SwiftDownload: NSObject, URLSessionDelegate, URLSessionDownloadDelegate {
+  /// The cache file of the (partial) download
+  private var finalSize: Int64?
+  private var progress: Int64 = 0
+  private var lastProgress: Int64 = -1
+  private var backChannel: FlutterMethodChannel
+  private var url: String?
+  private var headers = [String:String]()
+  private let urlHash: String
+  private var resumeData: Data?
+  private var task: URLSessionDownloadTask?
+  
+  init(urlHash : String, with binaryMessenger : FlutterBinaryMessenger) throws {
+    backChannel = FlutterMethodChannel(name: "fluttercommunity/flutter_downloader/\(urlHash)", binaryMessenger: binaryMessenger)
+    self.urlHash = urlHash
+
+    /// Parse meta file
+    var parseHeaders = false
+    let metaFile = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.appendingPathComponent("\(urlHash).meta")
+    let rawData = try String(contentsOf: metaFile, encoding: .utf8)
+    let lines = rawData.components(separatedBy:"\n")
+    for line in lines {
+      if line == "headers:" {
+        parseHeaders = true
+      } else {
+        let parts = line.split(separator: "=", maxSplits: 2)
+        let key = String(parts.first!)
+        let value = String(parts.last!)
+        if parseHeaders {
+          headers[key] = value
+        } else if key == "url" && !value.isEmpty {
+          url = value
+        // I think those fields are not relevant for iOS:
+        //} else if key == "filename" && !value.isEmpty {
+        //  filename = value
+        //} else if key == "etag" && !value.isEmpty {
+        // etag = value
+        //} else if key == "resumable" && !value.isEmpty {
+        //  resumable = value == "true";
+        //} else if key == "size" && !value.isEmpty {
+        //  finalSize = int.parse(value);
+        }
+      }
+    }
+  }
+  
+  func resume() {
+    let config = URLSessionConfiguration.background(withIdentifier: urlHash)
+    //config.timeoutIntervalForResource = Downloader.resourceTimeout
+    let urlSession = URLSession(configuration: config, delegate: self, delegateQueue: nil)
+    
+    if resumeData == nil {
+      print("Start download of \(url!)...")
```

Comment:
Let's create our simple custom logging class to not sprinkle `print()`s everywhere.

See an [example here](https://github.com/leancodepl/patrol/blob/patrol-v0.10.8/packages/patrol/ios/Classes/AutomatorServer/Logger.swift) of what I have in mind.

Reply:
I had in mind to remove those logging, but I can migrate to that instead too